src/Lecture3.hs

```diff
@@ -137,11 +151,16 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
+  (<>) :: List1 a -> List1 a -> List1 a
+  (List1 x1 xs1) <> (List1 x2 xs2) = List1  x1 (xs1 ++ x2:xs2)
 
 
 {- | Does 'List1' have the 'Monoid' instance? If no then why?
 
-instance Monoid (List1 a) where
+It can't have the Monoid instance, because
+there's no way to achive the equality:
+(List1 x xs) <> mempty == List1 x xs
+-- xs will always have one more element.
```

Comment:
That's correct! 👍🏻 
Or, in other words, there's no such thing as "empty non-empty list" 🙂